src/script/lua_api/l_mapgen.cpp

```diff
@@ -1006,6 +1006,32 @@ int ModApiMapgen::l_get_gen_notify(lua_State *L)
 }
 
 
+// get_deco_id(deconame)
+// returns the deco id as used in gennotify
+int ModApiMapgen::l_get_deco_id(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	const char *deco_str = lua_tostring(L, 1);
```

Comment:
`luaL_checkstring` is more appropriate here as it throws an error for a wrong use of this function or `nil` values

Reply:
Ok.

Reply:
Done.